modules/control/controller/lon_controller.cc

```diff
@@ -255,24 +255,35 @@ Status LonController::ComputeControlCommand(
       (chassis->gear_location() == canbus::Chassis::GEAR_REVERSE)
           ? -acceleration_cmd
           : acceleration_cmd;
-  if (FLAGS_use_preview_speed_for_table) {
-    calibration_value = control_interpolation_->Interpolate(
+  if (!FLAGS_use_acceleration) {
+    if (FLAGS_use_preview_speed_for_table) {
+      calibration_value = control_interpolation_->Interpolate(
         std::make_pair(debug->preview_speed_reference(), acceleration_lookup));
-  } else {
-    calibration_value = control_interpolation_->Interpolate(
+    } else {
+      calibration_value = control_interpolation_->Interpolate(
         std::make_pair(chassis_->speed_mps(), acceleration_lookup));
-  }
+    }
 
-  if (calibration_value >= 0) {
-    throttle_cmd = std::abs(calibration_value) > throttle_deadzone
+    if (calibration_value >= 0) {
+      throttle_cmd = std::abs(calibration_value) > throttle_deadzone
                        ? std::abs(calibration_value)
                        : throttle_deadzone;
-    brake_cmd = 0.0;
-  } else {
-    throttle_cmd = 0.0;
-    brake_cmd = std::abs(calibration_value) > brake_deadzone
+      brake_cmd = 0.0;
+    } else {
+      throttle_cmd = 0.0;
+      brake_cmd = std::abs(calibration_value) > brake_deadzone
                     ? std::abs(calibration_value)
                     : brake_deadzone;
+    }
+      cmd->set_throttle(throttle_cmd);
+      cmd->set_brake(brake_cmd);
+  } else {
+      cmd->set_acceleration(acceleration_cmd);
+    if (acceleration_cmd >= 0) {
+      cmd->set_throttle(acceleration_cmd);
```

Comment:
why you set acceleration_cmd to throttle or brake here, there is reversed field for acceleration command already   optional double acceleration = 10;

Reply:
https://github.com/ApolloAuto/apollo/blob/443318b4e123f2b87f7676828d4dd386fc6cece7/modules/control/proto/control_cmd.proto#L45

Reply:
> why you set acceleration_cmd to throttle or brake here, there is reserved field for acceleration command already optional double acceleration = 10;

Because there is a frame in canbus modules: vehicle_control.cc has defined the throttle and brake to implement cmd from control modules, so we do not change  the original structure.
[/modules/canbus/vehicle/vehicle_controller.cc](url)
I set the acceleration_cmd to throttle/brake just for keep original canbus （vehicle_controller.cc） structure.

Reply:
original canbus structure as follow：
[https://github.com/ApolloAuto/apollo/blob/master/modules/canbus/vehicle/vehicle_controller.cc](url)
`if (driving_mode_ == Chassis::COMPLETE_AUTO_DRIVE ||
      driving_mode_ == Chassis::AUTO_SPEED_ONLY) {
    Gear(control_command.gear_location());
    Throttle(control_command.throttle());
    Brake(control_command.brake());
    SetEpbBreak(control_command);
    SetLimits();
  }`